docs/manual/warnings.tex

```diff
@@ -711,12 +711,11 @@
 
 \item
   \code{cast:unsafe} (default: on) warn about unsafe casts that are not
-  checked at run time, as in \code{((@NonNull String) myref)}.  Such casts
-  are generally not necessary because of type refinement
-  (Section~\ref{type-refinement}).
+  checked at run time, as in \code{((@NonNull String) myref)}.
+  Such casts are unsafe, and they can break the property of the type system.
```

Comment:
I guess the original point was that manually adding downcasts is not necessary as flow-sensitive type refinement automatically refines the type if it is safe to do so.
The "break the property" sounds a bit odd to me. Can you try to reformulate the sentence.